pgjdbc/src/test/java/software/aws/rds/jdbc/postgresql/auth/AwsIamAuthenticationPluginTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+package software.aws.rds.jdbc.postgresql.auth;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+import org.postgresql.PGProperty;
+import org.postgresql.plugin.AuthenticationRequestType;
+import org.postgresql.util.PSQLException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.mockito.Mockito;
+
+import software.amazon.awssdk.regions.Region;
+
+import java.time.Instant;
+import java.util.Properties;
+
+public class AwsIamAuthenticationPluginTest {
+
+  private Properties properties;
+  private AwsIamAuthenticationPlugin targetPlugin;
+  private static final String TEST_USER = "testUser";
+  private static final String TEST_HOST = "test.user.us-east-2.rds.amazonaws.com";
+  private static final String TEST_TOKEN = "testToken";
+  private static final String GENERATED_TOKEN = "generatedToken";
+  private static final String CACHE_KEY = "us-east-2:test.user.us-east-2.rds.amazonaws.com:5342:testUser";
+
+  @BeforeEach
+  public void initTargetPlugin() throws PSQLException {
+    properties = new Properties();
+    properties.setProperty(PGProperty.USER.getName(), TEST_USER);
+    properties.setProperty(PGProperty.PG_HOST.getName(), TEST_HOST);
+    targetPlugin = new AwsIamAuthenticationPlugin(properties);
+  }
```

Comment:
AwsIamAuthenticationPlugin.tokenCache.clear()

We have to clear cache before each test

Reply:
:+1: